src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,767 @@
+         */
+        CYCLO_HEXANE
+    }
+    //</editor-fold>
+    //
+    //<editor-fold desc="Public Static Finals">
+    /**
+     * Name of the fragmenter.
+     */
+    public static final String ALGORITHM_NAME = "Alkyl structure";
+
+    /**
+     * Default option for maximum fragment length of carbohydrate chain, set to Methane.
+     */
+    public static final ChainFragmentLengthOption Chain_Fragment_LENGTH_OPTION_DEFAULT = ChainFragmentLengthOption.METHANE;
+    /**
+     * Default option for spiro carbon dissection.
+     */
+    public static final boolean DISSECT_SPIRO_CARBON_OPTION_DEFAULT = false;
+    /**
+     * Default option for ring dissection.
+     */
+    public static final boolean DISSECT_RINGS_OPTION_DEFAULT = false;
+    /**
+     * Default option for maximum size of preserved rings.
+     */
+    public static final PreserveRingMaxSizeOption Preserve_Ring_MAX_SIZE_OPTION_DEFAULT = PreserveRingMaxSizeOption.CYCLO_HEXANE;
+    /**
+     * Default option for ring system preservation.
+     */
+    public static final boolean PRESERVE_RING_SYSTEM_OPTION_DEFAULT = true;
+    /**
+     * Default option for maximum rings contained in preserved ring system.
+     */
+    public static final int PRESERVE_RING_SYSTEM_MAX_OPTION_DEFAULT = 1;
+    //</editor-fold>
+    //
+    //<editor-fold desc="Private Finals">
+    /**
+     * A property that has a constant fragment hydrogen saturation setting.
+     */
+    private final SimpleEnumConstantNameProperty fragmentSaturationSetting;
+    /**
+     * A property that has a constant name from the ChainFragmentLengthOption enum as value.
+     */
+    private final SimpleEnumConstantNameProperty chainFragmentLengthSetting;
+    /**
+     * Boolean property whether spiro carbons should be dissected.
+     */
+    private final SimpleBooleanProperty dissectSpiroCarbonSetting;
+    /**
+     * Boolean property whether rings should be dissected.
+     */
+    private final SimpleBooleanProperty dissectRingsSetting;
+    /**
+     * Enum property for maximum size of preserved rings.
+     */
+    private final SimpleEnumConstantNameProperty preserveRingMaxSizeSetting;
+    /**
+     * Boolean property whether ring systems should be preserved.
+     */
+    private final SimpleBooleanProperty preserveRingSystemSetting;
+    /**
+     * Integer property for maximum rings contained in preserved ring system.
+     */
+    private final SimpleIntegerProperty preserveRingSystemMaxSetting;
+    /**
+     * Map to store pairs of {@literal <setting name, tooltip text>}.
+     */
+    private final HashMap<String, String> settingNameTooltipTextMap;
+    /**
+     * All settings of this fragmenter, encapsulated in JavaFX properties for binding to GUI.
+     */
+    private final List<Property> settings;
+    /**
+     * Logger of this class.
+     */
+    private final Logger logger = Logger.getLogger(AlkylStructureFragmenter.class.getName());
```

Comment:
The logger should be a static final constant. The logger object you retrieve this way is static anyway.